pkg/service/auth.go

```diff
@@ -96,14 +118,22 @@ func (self *AuthService) successResponse(authResult AuthResult) *envoy_auth.Chec
 
 func (self *AuthService) deniedResponse(authResult AuthResult) *envoy_auth.CheckResponse {
 	code := authResult.Code
+
+	var httpCode envoy_type.StatusCode
+	if authResult.Status != 0 {
+		httpCode = authResult.Status
+	} else {
+		httpCode = statusCodeMapping[code]
+	}
```

Comment:
Could remove the else statement by assigning `httpCode` to `authResult.Status` and verifying if its 0 as suggested below 

```suggestion
        httpCode = authResult.Status
	if httpCode == 0 {
	   httpCode = statusCodeMapping[code]
	} 
```